pkg/ast/ast_type.go

```diff
@@ -27,10 +27,24 @@ type Type struct {
 	OfType   int
 }
 
+func (d *Document) BaseTypeNameBytes(ref int) ByteSlice {
+	if kind := d.Types[ref].TypeKind; kind == TypeKindNonNull || kind == TypeKindList {
+		return d.TypeNameBytes(d.Types[ref].OfType)
+	}
+	return d.TypeNameBytes(ref)
+}
+
 func (d *Document) TypeNameBytes(ref int) ByteSlice {
 	return d.Input.ByteSlice(d.Types[ref].Name)
 }
 
+func (d *Document) BaseTypeNameString(ref int) string {
+	if kind := d.Types[ref].TypeKind; kind == TypeKindNonNull || kind == TypeKindList {
+		return d.TypeNameString(d.Types[ref].OfType)
```

Comment:
Above comment applies here too. It also means you don't need to compare the type.